test/k4FWCoreTest/src/components/ExampleFunctionalMetadataConsumer.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2014-2024 Key4hep-Project.
+ *
+ * This file is part of Key4hep.
+ * See https://key4hep.github.io/key4hep-doc/ for further info.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "edm4hep/MCParticleCollection.h"
+
+#include "k4FWCore/Consumer.h"
+#include "k4FWCore/MetadataUtils.h"
+
+#include <string>
+#include <vector>
+
+struct ExampleFunctionalMetadataConsumer final : k4FWCore::Consumer<void(const edm4hep::MCParticleCollection& input)> {
+  ExampleFunctionalMetadataConsumer(const std::string& name, ISvcLocator* svcLoc)
+      : Consumer(name, svcLoc, KeyValues("InputCollection", {"MCParticles"})) {}
+
+  StatusCode initialize() override {
+    m_particleNum = k4FWCore::getParameter<int>("NumberOfParticles", this).value_or(0);
+    if (m_particleNum != 3) {
+      error() << "ExampleInt is not 3" << endmsg;
```

Comment:
One must print out the actual value of the parameter in all these error cases

Reply:
Done